containers/tomcat-7/src/main/java/org/mobicents/servlet/sip/startup/SipStandardContext.java

```diff
@@ -252,7 +254,20 @@ protected void prepareServletContext() throws LifecycleException {
 			if(proxyTimerServiceType != null && proxyTimerServiceType.equalsIgnoreCase("Standard")) {
                 proxyTimerService = new ProxyTimerServiceImpl(applicationName);
             } else if(proxyTimerServiceType != null && proxyTimerServiceType.equalsIgnoreCase("Default")) {
-                proxyTimerService = new DefaultProxyTimerService(applicationName);
+            	String strCorePoolSize = this.getServletContext().getInitParameter(TIMER_SERVICE_POOL_SIZE);
+            	if (strCorePoolSize != null && !strCorePoolSize.isEmpty()) {
+            		try {
+            			int CorePoolSize = Integer.parseInt(strCorePoolSize);
+            			proxyTimerService = new DefaultProxyTimerService(applicationName, CorePoolSize);
+            		}catch (NumberFormatException ex) {
+            			if (logger.isDebugEnabled()) {
+							logger.debug("Failed to parse timer service pool size, use default value.");
+						}
+            			proxyTimerService = new DefaultProxyTimerService(applicationName);

```

Comment:
i can see this will construct the timerService anyway, which is what I suggested.